daemon/service.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "common.h"
+
+void service_main_cleanup(void *ptr)
+{
+    struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITING;
+
+    debug(D_SYSTEM, "Cleaning up...");
+
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
+}
+
+/*
+ * The service thread.
+ */
+void *service_main(void *ptr)
+{
+    netdata_thread_cleanup_push(service_main_cleanup, ptr);
+    unsigned int sec = 0;
```

Comment:
We don't need this extra logic because `sleep` is cancelable. We can just sleep for `SERVICE_HEARTBEAT` seconds.

Reply:
Fixed.